File: ToasterLibrary/src/main/java/com/motion/toasterlibrary/productive/GenericRequestInterceptor.java
package com.motion.toasterlibrary.productive;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class GenericRequestInterceptor implements Interceptor {
    public static final String TAG = "GenericRequestInterceptor";

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request request = chain.request();
        System.out.println("INTERCEPT: " + request.url().url().toString());
        System.out.println("Method " + request.method());
        System.out.println("Headers " + request.headers());
        System.out.println("Body " + request.body());
        return chain.proceed(request);
    }
}

File: ToasterLibrary/src/main/java/com/motion/toasterlibrary/kotlin/GenericRequestInterceptor.kt
package com.motion.toasterlibrary.kotlin

import com.auth0.android.jwt.BuildConfig
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class GenericRequestInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()
        if (BuildConfig.DEBUG) {
            println("D INTERCEPT: " + request.url().url().toString())
            println("D Method " + request.method())
            println("D Headers " + request.headers())
            println("D Body " + request.body())
        }

        return chain.proceed(request)
    }

    companion object {
        const val TAG = "GenericRequestInterceptor"
    }
}